src/core/io/NewickConverter.cpp

```diff
@@ -52,45 +52,37 @@ Tree* NewickConverter::convertFromNewick(std::string const &n, bool reindex)
             {
                 trimmed += c;
             }
-
         }
-
     }
 
     // construct the tree starting from the root
     TopologyNode *root = createNode( trimmed, nodes, brlens );
 
     // set up the tree
-    t->setRoot( root, reindex );
+    t->setRoot( root, true );
+
+    // try to set node indices from attributes
+    t->tryReadIndicesFromParameters(true);
 
     // set the branch lengths
     for (size_t i = 0; i < nodes.size(); ++i)
     {
         t->getNode( nodes[i]->getIndex() ).setBranchLength( brlens[i] );
     }
 
-    // make all internal nodes bifurcating
-    // this is important for fossil trees which have sampled ancestors
-    t->makeInternalNodesBifurcating( reindex, true );
-
     // trees with 2-degree root nodes should not be rerooted
     t->setRooted( root->getNumberOfChildren() == 2 );
     
     // make this tree first a branch length tree
     // hence, tell the root to use branch lengths and not ages (with recursive call)
     root->setUseAges(false, true);
     
-    // if this tree is ultrametric, then we should use ages and transform so!
```

Comment:
be careful if you remove this. The above code uses "root->setUseAges(false, true);" where the first argument is false, but within the if clause it is set to true. Simply removing this seems dangerous to me.